spec/support/shared_examples/system/placements/edit_a_placement_wizard.rb

```diff
@@ -428,6 +437,21 @@ def when_i_select_year(year)
     choose year
   end
 
+  def and_the_provider_is_notified_they_have_been_assigned_to_the_placement(user)
+    and_the_provider_is_notified(user, "#{school.name} wants you to place a trainee with them")
+  end
+
+  def and_the_provider_is_notified_they_have_been_removed_from_the_placement(user)
+    and_the_provider_is_notified(user, "#{school.name} has removed you from a placement")
+  end
+
+  def and_the_provider_is_notified(user, subject)
+    ActionMailer::Base.deliveries.find do |delivery|
+      delivery.to.include?(user.email) &&
+        delivery.subject == subject
+    end
```

Comment:
Could the [`have_enqueued_mail` matcher](https://rspec.info/features/6-1/rspec-rails/matchers/have-enqueued-mail-matcher/) be helpful here?

We use it in similar tests such as this:

https://github.com/DFE-Digital/itt-mentor-services/blob/4a8aaa15f7e8db3d39b000d4a44faa6e40a16918/spec/services/placements/partnerships/notify/create_spec.rb#L29-L44

Reply:
I don't believe you can do that in a system rspec test.